### PR TITLE
Closes #166

### LINKED ISSUE #166: Add test to check for broken -hint chunks
-hint chunks containing broken code result in html files being corrupted in learnr.

### PR DESCRIPTION
**Pull request by soniamitchell**

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Package: RPiR
 Type: Package
 Title: Reproducible Programming in R
-Version: 0.53.0
+Version: 0.52.6
```
Comment (richardreeve) [inline DESCRIPTION:4]: This doesn't look ideal!

File: R/extract_chunks.R [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  labels <- lapply(chunks, function(x) 
+    strsplit(x[1], ",")[[1]][1] %>% 
+      gsub("^## ----", "", .) %>% 
+      gsub("-+$", "", .)) %>% 
+    unlist()
+  names(chunks) <- labels
+  
+  # Remove chunk header
+  chunks <- lapply(chunks, function(x) x[-1])
+}
```
Comment (richardreeve) [inline R/extract_chunks.R:32]: Need a newline :(

File: DESCRIPTION (hunk 1 above)
Comment (soniamitchell) [reply DESCRIPTION:4]: what happened here?!

**Comment by soniamitchell** [comment]
Closes #172

**Comment by soniamitchell** [comment]
Fixed "can't install" RPiR in CI. Added test for broken -hint chunks.

**Comment by richardreeve** [comment]
@soniamitchell By the way, you need to put the "closes" statement in the text of a comment to get automatic closing to work I think... though I see you then did it in one of the actuall commits instead.

**Comment by soniamitchell** [comment]
In a commit message? Sometimes I forget.. so I've been randomly commenting to make some kind of trail...

**Comment by richardreeve** [comment]
If you just say here that it closes #166 then when you merge the PR, #166 will be closed. You can tell because you get the little dots underlining the word "closes".

**Comment by richardreeve** [comment]
If you just say here that it closes it then when you merge the PR, then the issue will be closed. You can tell because you get the little dots underlining the word "closes".

Closes #166.

**Comment by richardreeve** [comment]
If you just say here that it closes #166 then when you merge the PR, #166 will be closed. You can tell because you get the little dots underlining the word "closes".

Edit: this doesn't seem to be working at the moment, maybe because this is already merged and #166 is closed!

**Comment by soniamitchell** [comment]
So I guess the comments are what is described in the documentation as the description then? 

Does my using _Closes_ in the commit message not work? Because it should...
https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by richardreeve** [comment]
Yes, it does work, and yes this text here is apparently a description - seems like a v bad name?! But in a commit it's not clear what it's referring to because git commits aren't tied to github. Whereas in a github PR, it's obvious it's referring to a github issue.

**Comment by soniamitchell** [comment]
I see 👍

**Comment by richardreeve** [comment]
People do put these statements in the commit messages all the time, but it feels to me like they are confusing git and GitHub when they do that.

**Comment by soniamitchell** [comment]
No, it does make sense. As one of those people that has never used git without GitHub, it took a moment to get my head around 🙈

**Pull request merged**